.circleci/config.yml

```diff
@@ -0,0 +1,143 @@
+# Circle CI configuration file
+# https://circleci.com/docs/
+
+version: 2
+
+
+###########################################
+# Define some common steps as YAML anchors.
+#
+
+apt-run:  &apt-install
+  name: Install apt packages
+  command: |
+    apt-get -qq update
+    apt-get install -y \
+      gcc \
+      g++ \
+      make
+
+env-run:  &env-setup
+  # Create the basic testing environment
+  # Mixing conda-forge and defaults in root is a recipe for trouble, so create
+  # a separate environment.
+  name: Setup conda environment
+  command: |
+    conda config --set always_yes yes --set changeps1 no --set show_channel_urls yes
+    conda create -n test-environment python=$PYTHON_VERSION
+    conda config --add channels conda-forge
+
+deps-run: &deps-install
+  name: Install Python dependencies
+  command: |
+    source activate test-environment
+    conda install --quiet \
+        numpy \
+        matplotlib \
+        scipy \
+        cython \
+        pillow \
+        proj4 \
+        pyshp \
+        shapely \
+        six \
+        requests \
+        pyepsg \
+        owslib \
+        $EXTRA_PACKAGES
+    conda install --quiet --file docs/doc-requirements.txt
+    conda list
+
+cp-run: &cp-install
+  name: Install Cartopy
+  command: |
+    source activate test-environment
+    pip install -ve .
+
+doc-run: &doc-build
+  name: Build documentation
+  command: |
+    source activate test-environment
+    make html
+  working_directory: docs
+
+doc-bundle-run: &doc-bundle
+  name: Bundle sphinx-gallery documentation artifacts
```

Comment:
Any good reason for doing this?

Reply:
Mostly to see what's wrong with the examples if the result comes out weird.